android/app/src/main/kotlin/coop/polypoly/polypod/Preferences.kt

```diff
@@ -24,27 +24,17 @@ class Preferences {
             edit.commit()
         }
 
-        fun getSeenInAppNotificationId(context: Context): Int =
-            getPrefs(context).getInt(seenInAppNotificationIdKey, 0)
-
-        fun setSeenInAppNotificationId(
-            context: Context,
-            seenNotificationId: Int
-        ) {
-            val edit = getPrefs(context).edit()
-            edit.putInt(seenInAppNotificationIdKey, seenNotificationId)
-            edit.commit()
+        fun getLastNotification(context: Context) = getPrefs(context).let {
+            Pair(
+                it.getInt(lastNotificationIdKey, 0),
+                it.getString(lastNotificationStateKey, null)
+            )
         }
 
-        fun getSeenPushNotificationId(context: Context): Int =
-            getPrefs(context).getInt(seenPushNotificationIdKey, 0)
-
-        fun setSeenPushNotificationId(
-            context: Context,
-            seenPushNotificationId: Int
-        ) {
+        fun setLastNotification(context: Context, id: Int, state: String) {
             val edit = getPrefs(context).edit()
-            edit.putInt(seenPushNotificationIdKey, seenPushNotificationId)
+            edit.putInt(lastNotificationIdKey, id)
```

Comment:
OK, let me see if I get this. If id is 0, you will allow the state to be whatever, but then in a second phase when you call `loadLastState`, it will have the effect of setting it to `ALL_SEEN`, that correct?

Reply:
Yes, but just in memory - it isn't actually being stored as `ALL_SEEN`, since no state changes happen and notifications with ID of `0` are ignored anyway. If _another_ notification is present, however, its ID will be stored along with its actual state (whenever it changes).